gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3075,10 +3081,19 @@ func doConvertType(store Store, last BlockNode, x *Expr, t Type) {
 //	convert to that even if right is a named type.
 //	case 2: isNamedConversion is called within evaluating make() or new()
 //	(uverse functions). It returns TypType (generic) which does have IsNamed appropriate
+//
+// This function also checks for the use of blank identifier "_" as a value or type,
+// which is not allowed. If both xt and t are nil, it panics with an appropriate error message.
 func isNamedConversion(xt, t Type) bool {
+	if xt == nil && t == nil {
+		panic("cannot use _ as value or type")
+	}
 	if t == nil {
 		t = xt
 	}
+	if xt == nil {
+		xt = t
+	}
```

Comment:
also this.

Reply:
I deleted this part as it seemed unnecessary. However, I added the suggested test as `typeassert10.gno`.

[1] https://github.com/gnolang/gno/pull/2699#discussion_r1845794927
[2] https://github.com/gnolang/gno/pull/2699#discussion_r1822719965